src/nsrt_learning/side_predicate_learning.py

```diff
@@ -394,59 +400,92 @@ def _sideline(self) -> List[PartialNSRTAndDatastore]:
-                # where delete effect variables also contribute to parameters.
-                opt_pred = Predicate("OPT-ARGS",
-                                     [a.type for a in pnad.option_spec[1]],
-                                     _classifier=lambda s, o: False)  # dummy
-                opt_vars_atom = frozenset(
-                    {LiftedAtom(opt_pred, pnad.option_spec[1])})
-                opt_objs_atom = frozenset(
-                    {GroundAtom(opt_pred, option.objects)})
-                unifies, obj_to_var_ent = utils.find_substitution(
-                    pnad.op.add_effects | opt_vars_atom,
-                    necessary_add_effects | opt_objs_atom)
-                obj_to_var = cast(ObjToVarSub, obj_to_var_ent)
-
-                if not unifies:  # we need to make add effects more specific
-                    del obj_to_var  # undefined; delete it for safety
-                    # We now try to make the PNAD's add effects more specific.
-                    # Since we have started with the most general operators,
-                    # and we are making them more specific, it is guaranteed
-                    # that such a grounding exists, otherwise our 1:1
-                    # option:operator assumption is violated.
-                    new_pnad = self._try_specializing_pnad(
-                        necessary_add_effects, pnad, segment)
-                    assert new_pnad is not None, "1:1 assumption violated"
-                    assert new_pnad.option_spec == pnad.option_spec
+                substitution_exists = False
+                # We start by checking if any of the PNADs associated with the
+                # demonstrated option are able to match this transition.
+                for pnad in pnads_for_option:
+                    ground_op = self._find_unification(necessary_add_effects,
+                                                       pnad, segment)
+                    if ground_op is not None:
+                        obj_to_var = dict(
+                            zip(ground_op.objects, pnad.op.parameters))
+                        if param_opt_to_nec_pnad.get(option.parent) is None:
+                            param_opt_to_nec_pnad[option.parent] = [pnad]
+                        substitution_exists = True
+                        break
+
+                # If we weren't able to find a substitution, we need to try
+                # specifize each of our PNADs.
+                if not substitution_exists:
+                    should_split_off_new_pnad = True
+                    for pnad in pnads_for_option:
+                        # We now try to make the PNAD's add effects more
+                        # specific. Since we have started with the most
+                        # general operators, and we are making them more
+                        # specific, it is guaranteed that such a
+                        # grounding exists, otherwise our 1:1
```

Comment:
can you go through and make sure all comments are fully up-to-date? (like this one, since we're no longer doing 1:1 stuff)